test/network_analysis/conservation_laws.jl

```diff
@@ -132,19 +160,82 @@ let
-    sprob.f(du1, u1, sprob.p, 1.0)
-    sprob2.f(du2, u2, sprob2.p, 1.0)
-    sprob3.f(du3, u3, sprob3.p, 1.0)
-    @test isapprox(du1, du2[istsidxs])
-    @test isapprox(du2[istsidxs], du3)
-    sprob.g(g1, u1, sprob.p, 1.0)
-    sprob2.g(g2, u2, sprob2.p, 1.0)
-    sprob3.g(g3, u3, sprob3.p, 1.0)
-    @test isapprox(g1, g2[istsidxs, :])
-    @test isapprox(g2[istsidxs, :], g3)
+    sprob1.f(du1, sprob1.u0, sprob1.p, 1.0)
+    sprob2.f(du2, sprob2.u0, sprob2.p, 1.0)
+    sprob3.f(du3, sprob3.u0, sprob3.p, 1.0)
+    @test du1 ≈ du2[ind_uidxs] ≈ du3
+    sprob1.g(g1, sprob1.u0, sprob1.p, 1.0)
+    sprob2.g(g2, sprob2.u0, sprob2.p, 1.0)
+    sprob3.g(g3, sprob3.u0, sprob3.p, 1.0)
+    @test g1 ≈ g2[ind_uidxs, :] ≈ g3
+end
+
+# Tests simulations for various input types (using X, rn.X, and :X forms).
+# Tests that conservation laws can be generated for system with non-default parameter types.
+let 
+    # Prepares the model.
+    rn = @reaction_network rn begin
+        @parameters kB::Int64 
+        (kB,kD), X + Y <--> XY
+    end
+    sps = species(rn)
+    @unpack kB, kD, X, Y, XY = rn
+
+    # Creates `ODEProblem`s using three types of inputs. Checks that solutions are identical.
+    u0_1 = [X => 2.0, Y => 3.0, XY => 4.0]
+    u0_2 = [rn.X => 2.0, rn.Y => 3.0, rn.XY => 4.0]
+    u0_3 = [:X => 2.0, :Y => 3.0, :XY => 4.0]
+    ps = (kB => 2, kD => 1.5)
+    oprob1 = ODEProblem(rn, u0_1, 10.0, ps; remove_conserved = true)
+    oprob2 = ODEProblem(rn, u0_2, 10.0, ps; remove_conserved = true)
+    oprob3 = ODEProblem(rn, u0_3, 10.0, ps; remove_conserved = true)
+    @test solve(oprob1)[sps] ≈ solve(oprob2)[sps] ≈ solve(oprob3)[sps]
 end
 
-### ConservedParameter Metadata Tests ###
+# Tests conservation laws in SDE simulation.
+let
+    # Creates `SDEProblem`s.
+    rn = @reaction_network begin
+        (k1,k2), X1 <--> X2
+    end
+    u0 = Dict([:X1 => 100.0, :X2 => 120.0])
+    ps = [:k1 => 0.2, :k2 => 0.15]
+    sprob = SDEProblem(rn, u0, 10.0, ps; remove_conserved = true)
+
+    # Checks that conservation laws hold in all simulations.
+    sol = solve(sprob, ImplicitEM(); seed)
+    @test sol[:X1] + sol[:X2] ≈ sol[rn.X1 + rn.X2] ≈ fill(u0[:X1] + u0[:X2], length(sol.t))
+end
+
+# Checks that the conservation law parameter's value can be changed in simulations.
+let 
+    # Prepares `ODEProblem`s.
+    rn = @reaction_network begin
+        (k1,k2), X1 <--> X2
+    end
+    osys = complete(convert(ODESystem, rn; remove_conserved = true))
+    u0 = [osys.X1 => 1.0, osys.X2 => 1.0]
+    ps_1 = [osys.k1 => 2.0, osys.k2 => 3.0]
+    ps_2 = [osys.k1 => 2.0, osys.k2 => 3.0, osys.Γ[1] => 4.0]
+    oprob1 = ODEProblem(osys, u0, 10.0, ps_1)
+    oprob2 = ODEProblem(osys, u0, 10.0, ps_2)
+
+    # Checks that the solutions generates the correct conserved quantities.
+    sol1 = solve(oprob1; saveat = 1.0)
+    sol2 = solve(oprob2; saveat = 1.0)
+    @test all(sol1[osys.X1 + osys.X2] .== 2.0)
+    @test all(sol2[osys.X1 + osys.X2] .== 4.0)
+end
```

Comment:
I also wanted to have a test checking
```julia
rn = @reaction_network begin
    @parameters k[1:2]
    @species X(t)[1:2]
    (k[1],k[2]), X[1] <--> X[2]
end
```
when conservation laws are eliminated (would be cool if it worked, but think there's an error). However, testing this took really long time. Might be related to the problem we saw earlier, but not sure.